doc/dune-files.rst

```diff
@@ -1413,6 +1413,37 @@ installed in. If the section above is documented as "with the executable bit
 set", they are installed with mode ``0o755`` (``rwxr-xr-x``); otherwise they are
 installed with mode ``0o644`` (``rw-r--r--``).
 
+Including files in the install stanza
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can include external files from the ``files`` and ``dirs`` fields of the
+install stanza:
+
+.. code:: scheme
+
+    (install
+     (files (include foo.sexp))
+     (section share))
+
+Here the file ``foo.sexp`` must contain a single S-expression list, whose
+elements will be included in the list of files or directories to install. That
+is, elements may be of the form:
+
+- ``<filename>``
+- ``(<filename> as <destination>)``
+- ``(include <filename>)``
+
+Included files may be generated by rules. Here is an example of a rule which
+generates a file by listing all the files in a subdirectory ``resources``:
+
+.. code:: scheme
+
+    (rule
+     (deps (source_tree resources))
+     (action
+      (with-stdout-to foo.sexp
+       (system "echo \"($(ls resources))\""))))
```

Comment:
Globbing requires less escaping, that might read better:

```suggestion
       (run echo "(" "resources/*" ")"))))
```

Reply:
also it's probably more correct since `ls resources` omits the `resources/` prefix

Reply:
The  `run` action won't work here because it runs the program directly rather than going via a shell, so globs aren't expanded. I agree that globbing is more readable and correct. Changing to `(system "echo '(' resources/* ')'")`